Extensions/TileMap/collision/TransformedTileMap.ts

```diff
@@ -0,0 +1,746 @@
+          if (definitionHitboxes) {
+            this.hitBoxes.length = definitionHitboxes.length;
+            for (
+              let polygonIndex = 0;
+              polygonIndex < this.hitBoxes.length;
+              polygonIndex++
+            ) {
+              const polygon = new gdjs.Polygon();
+              this.hitBoxes[polygonIndex] = polygon;
+              polygon.vertices.length = definitionHitboxes[polygonIndex].length;
+              for (
+                let vertexIndex = 0;
+                vertexIndex < polygon.vertices.length;
+                vertexIndex++
+              ) {
+                polygon.vertices[vertexIndex] = [0, 0];
+              }
+            }
+          }
+        }
+      }
+
+      /**
+       * @returns The tile definition from the tile set.
+       */
+      getDefinition(): TileMapHelper.TileDefinition {
+        return this.layer.tileMap.getTileDefinition(
+          this.layer._source.get(this.x, this.y)!
+        )!;
+      }
+
+      private _isHitboxesUpToDate() {
+        return (
+          this.affineTransformationUpToDateCount ===
+          this.layer.tileMap._transformationUpToDateCount
+        );
+      }
+
+      private _setHitboxesUpToDate() {
+        this.affineTransformationUpToDateCount = this.layer.tileMap._transformationUpToDateCount;
+      }
+
+      /**
+       * @returns The hitboxes of this tile in the scene basis.
+       */
+      getHitboxes(): Polygon[] {
+        if (this._isHitboxesUpToDate()) {
+          return this.hitBoxes;
+        }
+
+        const definition = this.getDefinition();
+        if (!definition) {
+          this._setHitboxesUpToDate();
+          // It should already be []
+          this.hitBoxes.length = 0;
+          return this.hitBoxes;
+        }
+        const tag = this.layer.tileMap.tag;
+        const definitionHitboxes = definition.getHitBoxes(tag);
+        if (!definitionHitboxes) {
+          this._setHitboxesUpToDate();
+          // It should already be []
+          this.hitBoxes.length = 0;
+          return this.hitBoxes;
+        }
+
+        const layerTransformation = this.layer.tileMap.getTransformation();
+        const width = this.layer.tileMap.getTileWidth();
+        const height = this.layer.tileMap.getTileHeight();
+
+        const tileTransformation =
+          TransformedCollisionTile.workingTransformation;
+        tileTransformation.setToTranslation(width * this.x, height * this.y);
+        if (this.layer.isFlippedHorizontally(this.x, this.y)) {
+          tileTransformation.flipX(width / 2);
+        }
+        if (this.layer.isFlippedVertically(this.x, this.y)) {
+          tileTransformation.flipY(height / 2);
+        }
+        if (this.layer.isFlippedDiagonally(this.x, this.y)) {
+          tileTransformation.flipDiagonally();
+        }
+        tileTransformation.preConcatenate(layerTransformation);
+
+        // TODO To handle tile map modification, update lengths.
```

Comment:
Not soon, at some point, we could try to allow events to modify the map. It would mean a per instance modification and, as there is several object for one tile map, it would need an identifier to tell which instance is the collision mask of a tile map instance.
I'm replacing it with "// The tile map can't change at runtime so the existing arrays can be reused safely."